fhir-server/src/main/java/com/ibm/fhir/server/registry/ServerRegistryResourceProvider.java

```diff
@@ -0,0 +1,215 @@
+/*
+ * (C) Copyright IBM Corp. 2020
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.ibm.fhir.server.registry;
+
+import static com.ibm.fhir.registry.util.FHIRRegistryUtil.getUrl;
+import static com.ibm.fhir.registry.util.FHIRRegistryUtil.isDefinitionalResource;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+
+import com.ibm.fhir.core.util.LRUCache;
+import com.ibm.fhir.model.resource.Resource;
+import com.ibm.fhir.model.resource.SearchParameter;
+import com.ibm.fhir.model.resource.StructureDefinition;
+import com.ibm.fhir.persistence.FHIRPersistence;
+import com.ibm.fhir.persistence.MultiResourceResult;
+import com.ibm.fhir.persistence.context.FHIRPersistenceContext;
+import com.ibm.fhir.persistence.context.FHIRPersistenceContextFactory;
+import com.ibm.fhir.persistence.helper.FHIRTransactionHelper;
+import com.ibm.fhir.persistence.helper.PersistenceHelper;
+import com.ibm.fhir.persistence.interceptor.FHIRPersistenceEvent;
+import com.ibm.fhir.persistence.interceptor.FHIRPersistenceInterceptor;
+import com.ibm.fhir.registry.resource.FHIRRegistryResource;
+import com.ibm.fhir.registry.resource.FHIRRegistryResource.Version;
+import com.ibm.fhir.registry.spi.FHIRRegistryResourceProvider;
+import com.ibm.fhir.search.context.FHIRSearchContext;
+import com.ibm.fhir.search.util.SearchUtil;
+
+public class ServerRegistryResourceProvider implements FHIRRegistryResourceProvider, FHIRPersistenceInterceptor {
+    public static final Logger log = Logger.getLogger(ServerRegistryResourceProvider.class.getName());
+
+    private final PersistenceHelper persistenceHelper;
+    private final Map<String, List<FHIRRegistryResource>> registryResourceMap = LRUCache.createLRUCache(1024);
```

Comment:
maybe overload the constructor to let the user pass in the cache size?

Reply:
This is used internally and is not intended to be configured / controlled by the user.